File: AVc/app/src/main/java/com/hfad/avc/Screens.java
package com.hfad.avc;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import com.hfad.avc.ui.contact.ContactFragment;
import com.hfad.avc.ui.main_fragment.MainFragment;
import com.hfad.avc.ui.namelist.NameListFragment;
import com.hfad.avc.ui.template.TemplateFragment;
import com.hfad.avc.ui.templateWrite.TemplateWriteFragment;

import ru.terrakok.cicerone.android.support.SupportAppScreen;

public class Screens {

    public static final class MainScreen extends SupportAppScreen {

        public MainScreen() {
            this.screenKey = getClass().getSimpleName();
        }

        @Override
        public Fragment getFragment() {
            MainFragment fragment = new MainFragment();
            return fragment;
        }
    }

    public static final class DbScreen extends SupportAppScreen {

        public DbScreen() {
            this.screenKey = getClass().getSimpleName();
        }

        @Override
        public Fragment getFragment() {
            NameListFragment fragment = new NameListFragment();
            return fragment;
        }
    }

    public static final class TemplateScreen extends SupportAppScreen {

        public TemplateScreen() {this.screenKey = getClass().getSimpleName();}

        @Override
        public Fragment getFragment() {
            TemplateFragment fragment = new TemplateFragment();
            return fragment;
        }
    }

    public static final class ConatctScreen extends SupportAppScreen {

        private final int congratulations;

        public ConatctScreen(int congratulations) {
            this.congratulations = congratulations;
            this.screenKey = getClass().getSimpleName();
        }

        @Override
        public Fragment getFragment() {
            ContactFragment fragment = new ContactFragment();
            Bundle args = new Bundle();
            args.putInt("congratulations", this.congratulations);
            fragment.setArguments(args);
            return fragment;
        }
    }
    public static final class TemplateWriteScreen extends SupportAppScreen {

        private final int templateId;

        public TemplateWriteScreen(int templateId) {
            this.templateId = templateId;
            this.screenKey = getClass().getSimpleName();
        }

        @Override
        public Fragment getFragment() {
            TemplateWriteFragment fragment = new TemplateWriteFragment();
            Bundle args = new Bundle();
            args.putInt("template_Id", this.templateId);
            fragment.setArguments(args);
            return fragment;
        }
    }

}

File: app/src/main/java/com/lastaurus/automatic_congratulations/cicerone/Screens.kt
package com.lastaurus.automatic_congratulations.cicerone

import android.os.Bundle
import androidx.fragment.app.FragmentFactory
import com.github.terrakok.cicerone.androidx.FragmentScreen
import com.lastaurus.automatic_congratulations.ui.congratulation.CongratulationFragment
import com.lastaurus.automatic_congratulations.ui.contact.ContactFragment
import com.lastaurus.automatic_congratulations.ui.main_fragment.MainFragment
import com.lastaurus.automatic_congratulations.ui.template.TemplateFragment


object Screens {

   @JvmStatic
   fun mainScreen() = object : FragmentScreen {
      override fun createFragment(factory: FragmentFactory) =
         MainFragment()
   }

   @JvmStatic
   fun contactScreen(contactId: Int) = object : FragmentScreen {
      override fun createFragment(factory: FragmentFactory) =
         ContactFragment().apply {
            val bundle = Bundle()
            bundle.putInt("contactId", contactId)
            arguments = bundle
         }
   }

   @JvmStatic
   fun templateScreen(templateId: Int) = object : FragmentScreen {
      override fun createFragment(factory: FragmentFactory) =
         TemplateFragment().apply {
            val bundle = Bundle()
            bundle.putInt("template_Id", templateId)
            arguments = bundle
         }
   }

   @JvmStatic
   fun congratulationScreen(congratulationId: Int) = object : FragmentScreen {
      override fun createFragment(factory: FragmentFactory) =
         CongratulationFragment().apply {
            val bundle = Bundle()
            bundle.putInt("congratulation_Id", congratulationId)
            arguments = bundle
         }
   }
}